From 5b5e4c29bd3357960db71a5dab139e22c596c7f1 Mon Sep 17 00:00:00 2001
From: Brian McGillion <bmg.avoin@gmail.com>
Date: Mon, 16 Jun 2025 08:20:38 +0400
Subject: [PATCH] tray-icon: Remove tray-icon

Not supported on ghaf. Needs to be tested if still applicable.

Signed-off-by: Brian McGillion <bmg.avoin@gmail.com>
---
 src/electron-main.ts | 71 +++++---------------------------------------
 1 file changed, 7 insertions(+), 64 deletions(-)

diff --git a/src/electron-main.ts b/src/electron-main.ts
index a00166c..63b3762 100644
--- a/src/electron-main.ts
+++ b/src/electron-main.ts
@@ -473,11 +473,10 @@ app.on("ready", async () => {
         // https://www.electronjs.org/docs/faq#the-font-looks-blurry-what-is-this-and-what-can-i-do
         backgroundColor: "#fff",
 
-        titleBarStyle: process.platform === "darwin" ? "hidden" : "default",
         trafficLightPosition: { x: 9, y: 8 },
 
         icon: global.trayConfig.icon_path,
-        show: false,
+        show: true,
         autoHideMenuBar: store.get("autoHideMenuBar"),
 
         x: mainWindowState.x,
@@ -492,91 +491,35 @@ app.on("ready", async () => {
             webgl: true,
         },
     });
-    void global.mainWindow.loadURL("vector://vector/webapp/");
-
-    if (process.platform === "darwin") {
-        setupMacosTitleBar(global.mainWindow);
-    }
 
     // Handle spellchecker
     // For some reason spellCheckerEnabled isn't persisted, so we have to use the store here
     global.mainWindow.webContents.session.setSpellCheckerEnabled(store.get("spellCheckerEnabled", true));
 
-    // Create trayIcon icon
-    if (store.get("minimizeToTray")) tray.create(global.trayConfig);
-
-    global.mainWindow.once("ready-to-show", () => {
+    global.mainWindow.webContents.once('did-finish-load',function() {
         if (!global.mainWindow) return;
         mainWindowState.manage(global.mainWindow);
 
         if (!argv["hidden"]) {
             global.mainWindow.show();
+            global.mainWindow.restore();
+            global.mainWindow.focus();
         } else {
             // hide here explicitly because window manage above sometimes shows it
             global.mainWindow.hide();
         }
     });
 
-    global.mainWindow.webContents.on("before-input-event", (event: Event, input: Input): void => {
-        const shouldWarnBeforeExit = store.get("warnBeforeExit", true);
-        const exitShortcutPressed =
-            input.type === "keyDown" && exitShortcuts.some((shortcutFn) => shortcutFn(input, process.platform));
-
-        if (shouldWarnBeforeExit && exitShortcutPressed && global.mainWindow) {
-            const shouldCancelCloseRequest =
-                dialog.showMessageBoxSync(global.mainWindow, {
-                    type: "question",
-                    buttons: [
-                        _t("action|cancel"),
-                        _t("action|close_brand", {
-                            brand: global.vectorConfig.brand || "Element",
-                        }),
-                    ],
-                    message: _t("confirm_quit"),
-                    defaultId: 1,
-                    cancelId: 0,
-                }) === 0;
-
-            if (shouldCancelCloseRequest) {
-                event.preventDefault();
-            }
-        }
-    });
+    global.mainWindow.loadURL("vector://vector/webapp/");
 
     global.mainWindow.on("closed", () => {
         global.mainWindow = null;
     });
     global.mainWindow.on("close", async (e) => {
-        // If we are not quitting and have a tray icon then minimize to tray
-        if (!global.appQuitting && (tray.hasTray() || process.platform === "darwin")) {
-            // On Mac, closing the window just hides it
-            // (this is generally how single-window Mac apps
-            // behave, eg. Mail.app)
-            e.preventDefault();
-
-            if (global.mainWindow?.isFullScreen()) {
-                global.mainWindow.once("leave-full-screen", () => global.mainWindow?.hide());
-
-                global.mainWindow.setFullScreen(false);
-            } else {
-                global.mainWindow?.hide();
-            }
-
-            return false;
-        }
+       // Close event handler
+       // Default behaviour is minimize to tray, that feature is removed since there is no tray support on Ghaf
     });
 
-    if (process.platform === "win32") {
-        // Handle forward/backward mouse buttons in Windows
-        global.mainWindow.on("app-command", (e, cmd) => {
-            if (cmd === "browser-backward" && global.mainWindow?.webContents.canGoBack()) {
-                global.mainWindow.webContents.goBack();
-            } else if (cmd === "browser-forward" && global.mainWindow?.webContents.canGoForward()) {
-                global.mainWindow.webContents.goForward();
-            }
-        });
-    }
-
     webContentsHandler(global.mainWindow.webContents);
 
     session.defaultSession.setDisplayMediaRequestHandler((_, callback) => {
-- 
2.49.0
